src/hooks/use-load-data-pages.ts

```diff
@@ -23,9 +23,33 @@ const shouldFetchAdditionalData = (
   );
 };
 
+const isMissingLayoutData = (layoutService: LayoutService): boolean => {
+  const { qHyperCube } = layoutService.layout;
+  // if qPivotDataPages is array
+  if (Array.isArray(qHyperCube.qPivotDataPages)) {
+    if (
+      // if qPivotDataPages not contains data
+      qHyperCube.qPivotDataPages.length === 0 ||
+      // if qPivotDataPages has data but it is empty
+      qHyperCube.qPivotDataPages[0].qData.length === 0 ||
+      // ???
```

Comment:
@cbt1 is this translates to fetch more on x axis? 
and below is fetch more as much as we can (within the boundary of a page height) in y axis?

Reply:
I would say no. We make an assumption there that we need to data on X and Y axis if any of the axeses are missing data